sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -838,6 +838,8 @@ case class RepartitionByExpression(
     numPartitions: Int) extends RepartitionOperation {
 
   require(numPartitions > 0, s"Number of partitions ($numPartitions) must be positive.")
+  require(partitionExpressions.nonEmpty, s"${getClass.getSimpleName} requires a non empty set of " +
```

Comment:
what was the behavior if `partitionExpressions` is empty?

Reply:
it was putting everything in the same partition